src/app/tracing/tracing.effects.ts

```diff
@@ -305,4 +308,57 @@ export class TracingEffects {
             return EMPTY;
         })
     ));
+
+    setStationPositionsAndLayoutMSA$ = createEffect(() => this.actions$.pipe(
+        ofType<tracingEffectActions.SetStationPositionsAndLayoutMSA>(
+            tracingEffectActions.TracingActionTypes.SetStationPositionsAndLayoutMSA
+        ),
+        withLatestFrom(this.store.pipe(select(tracingSelectors.getFclData))),
+        mergeMap(([action, state]) => {
+            try {
+                const oldStationPositionsAreEmpty = _.isEmpty(state.graphSettings.stationPositions);
+                const payloadPositionsAreEmpty = _.isEmpty(action.payload.stationPositions);
+
+                if ((oldStationPositionsAreEmpty) && (!payloadPositionsAreEmpty)) {
```

Comment:
This would be less redundant:

`this.store.dispatch(new tracingStateActions.SetStationPositionsAndLayoutSOA(action.payload));`
`if ((oldStationPositionsAreEmpty) && (!payloadPositionsAreEmpty)) {`
 ` ` `this.store.dispatch(new tracingEffectActions.RefreshReducedReferenceCopyMSA());`
`}`

Reply:
done!